jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1111,15 +1110,22 @@ private void parsedTrailer()
 
     private long convertContentLength(String valueString)
     {
-        try
-        {
-            return Long.parseLong(valueString);
-        }
-        catch (NumberFormatException e)
+        if (valueString == null || valueString.length() == 0)
+            throw new BadMessageException("Invalid Content-Length Value", new NumberFormatException());
+
+        long value = 0;
+        int length = valueString.length();
+
+
+        for (int i = 0; i < length; i++)
         {
-            LOG.trace("IGNORED", e);
-            throw new BadMessageException(HttpStatus.BAD_REQUEST_400, "Invalid Content-Length Value", e);
+            char c = valueString.charAt(i);
+            if (c < '0' || c > '9')
+                throw new BadMessageException("Invalid Content-Length Value", new NumberFormatException());
+
+            value = Math.addExact(Math.multiplyExact(value, 10), c - '0');
```

Comment:
@gregw before, `NumberFormatException` was caught here and converted, that's why I raised this comment.

Also, we can be more precise with the exception message saying "invalid content-length", rather than just an `ArithmeticException` with no message.

I'd prefer to catch and rethrow, but I can live as is.

Reply:
@sbordet I'd prefer to avoid the tiny cost of an extra try catch in critical path code to give a slightly better error message to a bad client.